daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -1482,6 +1518,74 @@ object Main {
         }
       }
 
+      case Some(conf.encodeEXI) => {
+        val encodeOpts = conf.encodeEXI
+        val channel = encodeOpts.output.toOption match {
+          case Some("-") | None => Channels.newChannel(STDOUT)
+          case Some(file) => new FileOutputStream(file).getChannel()
+        }
+        val output = Channels.newOutputStream(channel)
+
+        val inputStream = encodeOpts.infile.toOption match {
+          case Some("-") | None => STDIN
+          case Some(file) => {
+            val f = new File(file)
+            new FileInputStream(f)
+          }
+        }
+        val input = new InputSource(inputStream)
+
+        val exiFactory = {
+          if (encodeOpts.schema.isDefined)
+            getExiFactoryOpt(InfosetType.EXISA, encodeOpts.schema.toOption)
+          else
+            getExiFactoryOpt(InfosetType.EXI, encodeOpts.schema.toOption)
+        }
```

Comment:
Note that this has all been refactored in PR #873. We'll have to fix some merge conflicts depending on which is merged first. I think it should be fairly easy to update this to use the new stuff either way since it's just a matter of making the EXIFactory publicly available.

Reply:
I've merged that PR.

I think the changes you'll need are to make the `exiFactory` member of the `EXIInfosetHandler` public here:

https://github.com/apache/daffodil/blob/main/daffodil-cli/src/main/scala/org/apache/daffodil/InfosetTypes.scala#L676

Then you can call `getInfosetHandler` and access that `exiFactory` member. This only changes how you get the factory though, so I think everything else should be the same.